src/api/spec/controllers/webui/project_controller_spec.rb

```diff
@@ -713,6 +769,38 @@
     end
   end
 
+  describe 'POST #save_meta' do
+    before do
+      login user
+    end
+
+    context 'with an nonexistent project' do
+      let(:post_save_meta) { post :save_meta, params: { project: 'nonexistent_project' }, xhr: true }
+
+      it { expect{ post_save_meta }.to raise_error(Pundit::NotDefinedError) }
```

Comment:
I didn't write this spec but it does test the behaviour as it currently is implemented. Since `set_project_by_name` fails silently that means that Pundit will raise an error because we pass `nil` to `authorize`:

https://github.com/DavidKang/open-build-service/blob/project-controller-test/src/api/app/controllers/webui/project_controller.rb#L1168

Whether or not thats the desired behaviour is another question?